src/components/notebookRenderStrategy.tsx

```diff
@@ -121,4 +142,20 @@ export class NotebookRenderStrategy implements ExpandableNodeRenderStrategy {
 	private onChevronClick() {
 		this.expandNode();
 	}
+
+	private onExpand() {
+		if (this.notebook.needsToFetchChildren && this.notebook.apiUrl && this.globals.oneNoteDataProvider && !!this.globals.callbacks.onNotebookInfoReturned) {
+			this.globals.oneNoteDataProvider.getNotebookBySelfUrl(this.notebook.apiUrl, 5).then((notebook) => {
+				this.notebook.sections = notebook.sections
+				this.notebook.sectionGroups = notebook.sectionGroups
+			}).catch((apiError: OneNoteApi.RequestError) => {
+				this.notebook.apiHttpErrorCode = apiError.statusCode;
+			}).then(() => {
+				this.notebook.needsToFetchChildren = false;
+				if (!!this.globals.callbacks.onNotebookInfoReturned) {
```

Comment:
I think this should be moved above right after when getting the initial notebook succeeds

Reply:
Because of the way the picker is written, updating notebook properties doesn't rerender the picker but the callback can, so I added this here after, so it will rerender whether the getNotebooks succeeded or we enter the catch. The same is being done for sharedNotebooks

Reply:
Yes, it is clear this is needed, but can't it be called right after the API succeeds, instead of in a consecutive then block?

Reply:
I don't think it makes a difference in the case the api succeeds, but this will be called if the api fails as well.